scripts/test-util.js

```diff
@@ -0,0 +1,297 @@
+'use strict';
+const fs = require('fs');
+const os = require('os');
+const {spawnSync, spawn} = require('child_process');
+const net = require('net');
+
+const HZ_VERSION = '4.1.1';
+const HZ_TEST_VERSION = '4.1.1';
+const HAZELCAST_TEST_VERSION = HZ_TEST_VERSION;
+const HAZELCAST_VERSION = HZ_VERSION;
+const HAZELCAST_ENTERPRISE_VERSION = HZ_VERSION;
+const HAZELCAST_ENTERPRISE_TEST_VERSION = HZ_VERSION;
+const HAZELCAST_RC_VERSION = '0.8-SNAPSHOT';
+const SNAPSHOT_REPO = 'https://oss.sonatype.org/content/repositories/snapshots';
+const RELEASE_REPO = 'http://repo1.maven.apache.org/maven2';
+const ENTERPRISE_RELEASE_REPO = 'https://repository.hazelcast.com/release/';
+const ENTERPRISE_SNAPSHOT_REPO = 'https://repository.hazelcast.com/snapshot/';
+const ON_WINDOWS = os.platform() === 'win32';
+const ENV_VARIABLE_SEPARATOR = ON_WINDOWS ? ';' : ':';
+const HAZELCAST_ENTERPRISE_KEY = process.env.HAZELCAST_ENTERPRISE_KEY ? process.env.HAZELCAST_ENTERPRISE_KEY : '';
+
+let MOCHA_COMMAND;
+let REPO;
+let ENTERPRISE_REPO;
+let CLASSPATH;
+
+let testType;
+let rcProcess;
+let mochaProcess;
+
+const downloadDependencies = () => {
```

Comment:
nit: I'd suggest to keep this part in `download-rc.js` script as it seems to be self-contained.

Reply:
It was a little hard due to CLASSPATH variable usage in test-runner.js, but I achieved this by extracting the RC parameters to a new file.